main.js

```diff
@@ -0,0 +1,284 @@
+                     * {Buffer} content: contents of `filename`
+                     */
+                    let matter = docmatter(content.toString('UTF-8'));
+
+                    return serverTemplate(
+                        Object.assign({
+                            body: Markdown.render(matter.body)
+                        }, globalContext, yaml.safeLoad(matter.header))
+                    );
+                }).then(html => fse.writeFile(path.join(outDir, dirname, filename.replace(/\.md$/i,".html")), html))
+                
+            )
+        return fse.mkdirp(path.join(outDir, dirname)).then(Promise.all(promises));
+    })
+}
+
+/**
+ * Formats a date "Month DD, YYYY"
+ * @param {Date} date
+ * @returns {String} 
+ */
+function formatDate(date) {
+    const monthNames = [
+        "January", "February", "March",
+        "April", "May", "June", "July",
+        "August", "September", "October",
+        "November", "December"
+      ];
+    return `${monthNames[date.getMonth()]} ${date.getDate()}, ${date.getFullYear()}`;
+}
+
+/**
+ * Generates an index of a directories PDF files, and copies them to the HTML folder.
+ * @param {string} dirname Directory to find .PDF files in
+ * @param {string} filename Filename of this page
+ * @param {Handlebars.TemplateDelegate} wrapperTemplate Handlebars template to apply to the whole page
+ * @param {Handlebars.TemplateDelegate} minuteTemplate Template to use on the listing
+ * @param {Object} globalContext Context to use for every page
+ */
+function minutes(dirname, filename, wrapperTemplate, minuteTemplate, globalContext) {
+    return fse.readdir(dirname).then(listing => {
+        let minutes = listing.filter(l => /\.pdf$/.test(l))
+                        .map(fn => {
+                            let m = /^(\d{4}-\d\d-\d\d)-([a-z0-9 ]+)\.pdf$/i.exec(fn);
+
+                            return {
+                                url: path.join(dirname, fn),
+                                meeting: m[2],
+                                date: m[1],
+                                date_t: new Date(m[1])
+                            }
+                        }).sort((f1, f2) => Math.sign(f1.date_t - f2.date_t));
+                        // Has the potential to have meeting on the same date in the wrong order
+                        // Cross that bridge when etc
+
+        fse.writeFile(path.join(outDir, filename), 
+            wrapperTemplate(Object.assign({
+                body: minuteTemplate(minutes),
+                title: "Minutes"
+            }, globalContext))
+        ).then(fse.copy(dirname, path.join(outDir, dirname)))
+    })
+}
+
+/**
+ * @param {Object} results any previous results that this object carries on
+ * @param {String} dirname directory to search for news
+ */
+function readNews(results, dirname='news') {
+    const re_date = /^(\d{4})-([01]\d)-([0-3]\d)/;
+    return new Promise((resolve, reject) => {
+        fse.readdir(dirname).then(listing => {
```

Comment:
Less likely to occur than with minutes above, however there should be some form of error handling for this `readdir`.